crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -219,56 +343,83 @@ fn parse_entries_numpy(content: &str) -> Vec<QualifiedName> {
     entries
 }
 
+/// An individual documentable statement in a function body.
+#[derive(Debug)]
+struct Entry {
+    range: TextRange,
+}
+
+impl Ranged for Entry {
+    fn range(&self) -> TextRange {
+        self.range
+    }
+}
+
 /// An individual exception raised in a function body.
 #[derive(Debug)]
-struct Entry<'a> {
+struct ExceptionEntry<'a> {
     qualified_name: QualifiedName<'a>,
     range: TextRange,
 }
 
-impl Ranged for Entry<'_> {
+impl Ranged for ExceptionEntry<'_> {
     fn range(&self) -> TextRange {
         self.range
     }
 }
 
-/// The exceptions raised in a function body.
+/// A summary of documentable statements from the function body
 #[derive(Debug)]
 struct BodyEntries<'a> {
-    raised_exceptions: Vec<Entry<'a>>,
+    returns: Vec<Entry>,
+    raised_exceptions: Vec<ExceptionEntry<'a>>,
 }
 
-/// An AST visitor to extract the raised exceptions from a function body.
+/// An AST visitor to extract a summary of documentable statements from a function body.
 struct BodyVisitor<'a> {
-    raised_exceptions: Vec<Entry<'a>>,
+    returns: Vec<Entry>,
+    raised_exceptions: Vec<ExceptionEntry<'a>>,
     semantic: &'a SemanticModel<'a>,
 }
 
 impl<'a> BodyVisitor<'a> {
     fn new(semantic: &'a SemanticModel) -> Self {
         Self {
+            returns: Vec::new(),
             raised_exceptions: Vec::new(),
             semantic,
         }
     }
 
     fn finish(self) -> BodyEntries<'a> {
         BodyEntries {
+            returns: self.returns,
             raised_exceptions: self.raised_exceptions,
         }
     }
 }
 
 impl<'a> Visitor<'a> for BodyVisitor<'a> {
     fn visit_stmt(&mut self, stmt: &'a Stmt) {
-        if let Stmt::Raise(ast::StmtRaise { exc: Some(exc), .. }) = stmt {
-            if let Some(qualified_name) = extract_raised_exception(self.semantic, exc.as_ref()) {
-                self.raised_exceptions.push(Entry {
-                    qualified_name,
-                    range: exc.as_ref().range(),
-                });
+        match stmt {
```

Comment:
This is not specific to your change but won't this visitor traverse into nested functions?

```python
def test():
	def nested():
		return 5

	print("I never return")
```